homeassistant/components/epson/media_player.py

```diff
@@ -104,6 +108,16 @@ def __init__(self, projector, name, unique_id):
         self._state = None
         self._unique_id = unique_id
 
+    async def set_unique_id(self):
+        """Set unique id for projector config entry."""
+        _LOGGER.debug("Setting unique_id for projector.")
+        if not self._unique_id:
+            if await self._projector.get_serial_number():
```

Comment:
Combine this row with the row above and invert the logic.

Reply:
We want to save the serial number and update the config entry unique_id using `hass.config_entries.async_update_entry`.